task/prefetch-dependencies-oci-ta/0.1/prefetch-dependencies-oci-ta.yaml

```diff
@@ -398,6 +411,9 @@ spec:
       image: quay.io/redhat-appstudio/cachi2:0.15.0@sha256:b141cb5cf4d98e6c5f668f1fe172e1d68f2a44ac1027403fbcff94ce1e68185d
       script: |
         #!/bin/bash
+
+        set -euo pipefail
```

Comment:
Oof, I just saw the `unregister-rhsm` step at the end. This means the prefetch task will only unregister itself if it succeeds, otherwise it will leave a registration hanging around in the system.

But exiting on failure properly probably doesn't make that problem much worse than it already was

Reply:
questionable is, how much of hanging are we talking about in ephemeral tasks

Reply:
I can rework it to hard way using an internal state

Reply:
by hard way I mean to `trap` errror, record it and skip all tasks except unregister